### DISCUSSION TITLE
get_flexi_tvis_affine(tvis_hdr, nii_aff)

### DISCUSSION THREAD

**Discussion by RosaPetit** [dipy/dipy#2636] [Preprocessing]
## Description
Is there a replacement for this function "get_flexi_tvis_affine"?

I'm trying to get code to work that uses this function: "get_flexi_tvis_affine", but it has already been removed (1.0.0) from new versions of Dipy. Is there any way to emulate it with the new version of Dipy 1.5?

**Comment by skoudoro** [comment]
Hi @RosaPetit,

Not sure there is an exact replacement for this function.

Can you provide the link or the script of what you are trying to achieve?

**Reply by RosaPetit** [reply]
Hi! @skoudoro 

Thanks for the answer!

You can find the script in the link below: 

https://pubmed.ncbi.nlm.nih.gov/31544062/

**Comment by skoudoro** [comment]
if you look at the `path_length` function at this link: https://github.com/dipy/dipy/blob/fe876e59d265f8785d5d6222b0ef522d9d2936e0/dipy/tracking/utils.py#L956

you can see that she is using  `_to_voxel_coordinates` now. So I suppose, I can use the same now to replace `get_flexi_tvis_affine`

you can import it like this `from dipy.tracking._utils import (_mapping_to_voxel, _to_voxel_coordinates)`

and look here at this link for more information of this function: https://github.com/dipy/dipy/blob/fe876e59d265f8785d5d6222b0ef522d9d2936e0/dipy/tracking/_utils.py

I hope it helps.
